packages/react-component-library/src/components/TopLevelNavigation/Masthead/Masthead.stories.tsx

```diff
@@ -322,3 +325,21 @@ WithClassificationBar.args = {
   classificationBar: <ClassificationBar />,
 }
 WithClassificationBar.storyName = 'Classification bar'
+
+const StyledClientComponent = styled.div`
+  flex: 1;
+  display: flex;
+  align-items: center;
+  justify-content: end;
+  padding-right: ${spacing('2')};
+`
+
+export const WithCustomClientComponent = Default.bind({})
+WithCustomClientComponent.args = {
+  ...Default.args,
+  rightSlot: (
+    <StyledClientComponent>
+      <TextE>Arbitrary text</TextE>
+    </StyledClientComponent>
+  ),
+}
```

Comment:
Change to `export const RightSlot` and add the following so the story title case is correct:

```
RightSlot.storyName = 'Right slot'
```